### PR TITLE
Adds ability to disable moving or zooming a floor plan.

### PR DESCRIPTION
**Pull request by digitaldan**
Signed-off-by: digitaldan <dan@digitaldan.com>

### CODE REVIEW

**Comment by digitaldan** [comment]
Hi @ghys ! I though i would dip my toe into vue.js by implementing a small feature.  This ended up being a pretty good learning experience as i ended needing to understand all sorts concepts around computed/watched properties, component lifecycle, etc...   Anyways, curious if i managed to do this right or not?

**Comment by digitaldan** [comment]
zoomDrag

**Comment by ghys** [comment]
Awesome @digitaldan, it looks fine to me at first sight! Normally zooming & panning would be an useful feature to make sure the floor plans are usable on all screens (i.e. on a phone you would probably have to zoom a bit), but for fixed aspect ratios it's a good idea to offer the option.

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue [hunk 1]
```diff
@@ -101,6 +102,12 @@ export default {
           type: 'INTEGER',
           description: 'The width of the image (by default 1000 pixels). Please specify if the image is not square to compute the aspect ratio'
         },
+        {
+          name: 'noZoomOrDrag',
+          label: 'Disable Ability to Zoom, Drag or Move',
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]: Isn't dragging and moving pretty much the same thing in this context?

```suggestion
          label: 'Disable Zooming & Dragging',
```

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue [hunk 2]
```diff
@@ -143,20 +150,34 @@ export default {
...
+      } : {})
+    },
   },
   mounted () {
-    this.$refs.map.mapObject.fitBounds(this.bounds)
+    this.fitMapBounds()
+  },
+  watch: {
+    'config.noZoomOrDrag' : function (val) {
+      this.refeshMap()
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]:
```suggestion
      this.refreshMap()
```

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue [hunk 3]
```diff
@@ -143,20 +150,34 @@ export default {
...
+      zoomControl:false
+      } : {})
+    },
   },
   mounted () {
-    this.$refs.map.mapObject.fitBounds(this.bounds)
+    this.fitMapBounds()
+  },
+  watch: {
+    'config.noZoomOrDrag' : function (val) {
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]: As general advice,,watchers are normally discouraged - but here it makes sense in this case :)
Though we'd probably check in a debugger if it's called when the map is first rendered (even if the option isn't set), in which case we wouldn't need the `fitMapBounds()` call in `mounted`; the goal would be to call it only once if the config doesn't change.

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue [hunk 4]
```diff
@@ -169,6 +190,15 @@ export default {
       return 'oh-plan-marker'
     },
     onMarkerUpdate () {
+    },
+    fitMapBounds() {
+      this.$refs.map.mapObject.fitBounds(this.bounds)
+    },
+    refeshMap() {
```
Comment (ghys) [inline bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]:
```suggestion
    refreshMap() {
```

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue (hunk 2 above)
Comment (digitaldan) [reply bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]: Uhg, typo AND i copied and pasted it !

File: bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue (hunk 3 above)
Comment (digitaldan) [reply bundles/org.openhab.ui/web/src/components/widgets/plan/oh-plan-page.vue]: Just checked and it's only calling fitMapBounds when the config option changes

**Comment by digitaldan** [comment]
Thanks @ghys for reviewing !

**Comment by ghys** [comment]
Great, let me know when you're finished with it (by removing the WIP prefix in the title), LGTM!
Thanks for your participation in this effort 😆

**Comment by digitaldan** [comment]
Done!

**Pull request merged**